File: carshop-admin/src/api/category.ts
import { apiClient, unwrap } from './client';

export interface Category {
  id: number;
  name: string;
  icon_url: string;
  sort: number;
  created_at: string;
}

export interface CategoryInput {
  name: string;
  icon_url: string;
  sort: number;
}

export const categoryApi = {
  listPublic: () => unwrap<Category[]>(apiClient.get('/categories')),
  listAdmin: () => unwrap<Category[]>(apiClient.get('/admin/categories')),
  create: (body: CategoryInput) =>
    unwrap<Category>(apiClient.post('/admin/categories', body)),
  update: (id: number, body: Partial<CategoryInput>) =>
    unwrap<Category>(apiClient.put(`/admin/categories/${id}`, body)),
  remove: (id: number) =>
    unwrap<{ id: number; deleted: boolean }>(apiClient.delete(`/admin/categories/${id}`)),
};

File: carshop-android/app/src/main/java/com/carshop/android/data/dto/Category.kt
package com.carshop.android.data.dto

import com.squareup.moshi.Json

data class Category(
    val id: Int,
    val name: String,
    @Json(name = "icon_url") val iconUrl: String,
    val sort: Int,
    @Json(name = "created_at") val createdAt: String,
)
